engine/consensus/dkg/processor/engine.go

```diff
@@ -136,36 +125,70 @@ func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
 func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 	switch v := event.(type) {
 	case msg.DKGMessage:
-		return e.onMessage(originID, v)
+		return e.onPrivateMessage(originID, v)
 	default:
 		return fmt.Errorf("invalid event type (%T)", event)
 	}
 }
 
-func (e *Engine) onMessage(originID flow.Identifier, msg msg.DKGMessage) error {
-	if currentDKG := e.GetDKGInstanceID(); currentDKG != msg.DKGInstanceID {
-		return fmt.Errorf("wrong DKG instance ID. Got %v, want %v", msg.DKGInstanceID, currentDKG)
-	}
-	if msg.Orig >= len(e.committee) || msg.Orig < 0 {
-		return fmt.Errorf("origin id out of range: %d", msg.Orig)
+func (e *Engine) onPrivateMessage(originID flow.Identifier, msg msg.DKGMessage) error {
+	err := e.checkMessageInstanceAndOrigin(msg)
+	if err != nil {
+		return err
 	}
+	// check that the message's origin matches the sender's flow identifier
 	nodeID := e.committee[msg.Orig]
 	if !bytes.Equal(nodeID[:], originID[:]) {
 		return fmt.Errorf("OriginID (%v) does not match committee member %d (%v)", originID, msg.Orig, nodeID)
 	}
-	e.log.Debug().Msgf("forwarding Message to controller")
+	e.log.Debug().Msgf("forwarding private message to controller")
 	e.msgCh <- msg
 	return nil
 }
 
+// fetchBroadcastMessages calls the DKG smart contract to get missing DKG
+// messages for the current epoch, and forwards them to the msgCh.
+// It should be called with the ID of block whose seal is finalized.
```

Comment:
Another note on usage here: the caller will also need to ensure that the block is within the appropriate DKG phase. 

Suppose `B` is the final block of DKG phase 1.

@tarakby and I discussed a race condition around this today where the process periodically calling `fetchBroadcastMessages` uses a block which is a child of `B` (ie. within DKG phase 2) before the process which handles the phase transition indicates to the `DKGProcessor` that we have transitioned to phase 2. It would be good to add a sanity check to the "periodic fetching" process to ensure the blocks it uses as reference are strictly within the current DKG phase.